app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: %i[edit update]
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @user.update(user_params)
+      flash.notice = "User '#{@user.username}' Updated!"
+      redirect_to edit_user_path
+    else
+      redirect_to edit_user_path
+      # flash.notice = "Some errors found try again!!!"
+      flash.notice = @user.errors.full_messages
+    end
```

Comment:
This should `render edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))